libraw/libraw.h

```diff
@@ -270,7 +270,7 @@ class DllDef LibRaw
 
   int FC(int row, int col)
   {
-    return (imgdata.idata.filters >> (((row << 1 & 14) | (col & 1)) << 1) & 3);
+    return (imgdata.idata.filters >> ((((row > 0 ? row : -row) << 1 & 14) | ((col > 0 ? col : -col) & 1)) << 1) & 3);
```

Comment:
this change may result in great slow down if FC() is used on each image pixel (not recommended, but possible).

Both original code and your change will/may result into incorrect output: original code because of UB in negative shift, your change assumes mirrored CFA that is not true.

Reply:
BTW, I could not understand what 'Issue 22900' file illustrates: I've added conditional breakpoint for (row < 0 || col < 0) in FC() and it was not hit.

Reply:
I understand, it seems the code that routinely calls FC with negative values relies on this property mirroring to be true (https://github.com/LibRaw/LibRaw/blob/4a39decd113ca1c08db5b5ed1c9e96f6867e5f0e/src/demosaic/misc_demosaic.cpp#L149). I worry as negative shifts are not guaranteed behaviour in the standard, not sure if there’s a better solution

Reply:
Strange it’s not getting hit, you can see the above function will always call FC(-1, -1) for example.

Reply:
lin_interpolate should be fixed by adding fixed offsets for fcol() params

Reply:
Here is the fix: https://github.com/LibRaw/LibRaw/commit/5f5046b62381ea44bd7b4f250aee387197987702

Reply:
and another one for vng_interpolate(): https://github.com/LibRaw/LibRaw/commit/94a3fa113951a1b8403e55220e61194271853de7